TakingITGlobal/commit2act
# [TIG-779] User status for conditional routing

## Ticket(s)

* [TIG-779](https://app.clickup.com/t/9009201449/TIG-779) (focused sub-task)
* [TIG-734](https://app.clickup.com/t/9009201449/TIG-734) (parent task)

## Description

This is a sibling of [TIG-778,](https://github.com/TakingITGlobal/commit2act/pull/113) following up on the previous PR's updates to complete the routing and user status implementation!

This PR establishes a consistent, app-wide check of the user's logged-in and admin status at the `Router` level, which is then used to redirect the browser if a user's status does not match the requirements of the requested page. The currently-in-production method relies more on selectively rendering routes based on user status - which makes redirects and auth gates much tougher to manage consistently. 

The app does have some minor user status info in the global Redux store, but this approach aims to complement that without relying on the extra provider wrappers and delicate global state setup. By referring to our AWS Amplify source-of-truth throughout the routing, we know the user session is up-to-date and not stale. This will be critical for controlling our new onboarding/sign up flow and ensuring that each step is updated with the user's selections and entries from the previous steps.

**Note:** We're still using the `REACT_APP_SIGNUP_FLOW` environment variable, so you won't see this new routing approach render locally unless that option is changed to `onboard` from `simple`.

## Testing

1. Pull down the branch and install (`npm install`)
2. Run the test suite (`npm run test`) and confirm everything passes
3. Change the `REACT_APP_SIGNUP_FLOW` value in the `.env` file to `onboard` from `simple`
4. Run the local app (`npm run start`)
5. From the login page, confirm selecting "create account" redirects to the new Sign Up page 
6. Enter a random sub-page as the URL and confirm you are redirected to the login page
7. Login with a team or admin user and confirm all expected pages render and menu items appear
8. Navigate to the first onboarding page (`/onboard/details`) and confirm it renders while you are logged in
9. Step through the onboarding pages using the back/next navigation and confirm they all render and redirect as expected
10. Navigate throughout pages in the app, refreshing the page on some and confirming your login status and rendered page(s) work as expected
11. Log out and repeat step 10 with public pages (Login, Signup)
